File: 2424-longest-uploaded-prefix/2424-longest-uploaded-prefix.cpp
class LUPrefix {
public:
    unordered_map<int,int> map;
    int i=1;
    int sz=0;
    LUPrefix(int n) {
        sz=n;
    }
    
    void upload(int video) {
        map[video]=1;
        if(video==i)i++;
    }
    
    int longest() {
        while(map[i])i++;
        return i-1;
    }
};

File: 2424-longest-uploaded-prefix/2424-longest-uploaded-prefix.java
class LUPrefix {
    Set<Integer> set;
    int max=0;
    public LUPrefix(int n) {
        set=new HashSet<>();
    }
    public void upload(int video) {
        set.add(video);
        while(set.contains(max+1)) max++;
    }
    public int longest() {
        return max;
    }
}